### PR TITLE
TestCases added

### PR DESCRIPTION
**Pull request by ufjena**
Can you take a look at the code when you have time?
I have a few questions then.

### CODE REVIEW

File: config/neodymium.properties [hunk 1]
```diff
@@ -9,8 +9,10 @@
 # To run any show case the Chrome WebDriver must be installed before.
 # See: https://github.com/Xceptance/neodymium-library/wiki/How-to-set-up-a-WebDriver
 # Configure 
-# neodymium.webDriver.chrome.pathToDriverServer = c:/path/to/chromedriver.exe
+neodymium.webDriver.chrome.pathToDriverServer = C:/Users/uwe/Documents/WebDriver/chromedriver.exe
```
Comment (occupant23) [inline config/neodymium.properties]: please use the dev-neodymium.properties file to configure your local dev environment

File: config/neodymium.properties [hunk 2]
```diff
@@ -23,7 +25,7 @@ neodymium.locale = en_US
 # - SelenideBasicAuthentication
 # - NeodymiumLocalProxyCertificateAuthentication
 # - NeodymiumLocalProxyCertificateAuthenticationHost
-neodymium.localproxy = false
+neodymium.localproxy = true
```
Comment (occupant23) [inline config/neodymium.properties]: please revert this change to keep the original setup running

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.ex.ElementNotFound;
+
+import io.qameta.allure.Description;
+
+public class FixedRandomTest {
+	private Random random;
+
+	protected static final Logger LOGGER = LoggerFactory.getLogger(RandomJobOffersTest.class);
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: Move the LOGGER example to another (its own) package/test. We try to concentrate on one thing only when preparing a showcase.

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+	@Test
+	@Description(value = "Showcase for fixec random Test.")
+	public void testRandomJobs() {
+		// open the demo page and prepare it for the test
+		openJobPage();
+
+		// count the job offers in the list
+//		.jobs-plain-list>li>a
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: remove unneeded comments (like unused code and selectors)

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.ex.ElementNotFound;
+
+import io.qameta.allure.Description;
+
+public class FixedRandomTest {
+	private Random random;
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: No need for a class field. Use a local variable instead to keep this simple.

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+		try {
+			$(".btn-link").shouldBe(visible);
+		} catch (ElementNotFound e) {
+			overlayIsVisible = false;
+		}
+
+		if (overlayIsVisible) {
+			$(".btn-link").click();
+			$(".btn-link").shouldBe(hidden);
+			Selenide.refresh();
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: a refresh should not be needed at this place

File: src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+import static com.codeborne.selenide.Condition.attribute;
+import static com.codeborne.selenide.Condition.hidden;
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Condition.textCaseSensitive;
+import static com.codeborne.selenide.Condition.exactText;
+import static com.codeborne.selenide.Condition.text;
+import static com.codeborne.selenide.Condition.matchText;
+import static com.codeborne.selenide.CollectionCondition.sizeGreaterThanOrEqual;
+//import static com.codeborne.selenide.ElementsCollection;
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java]: remove commented imports

File: src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+import com.xceptance.neodymium.util.Neodymium;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RandomJobOffersTest {
+
+	private Random random;
+
+	protected static final Logger LOGGER = LoggerFactory.getLogger(RandomJobOffersTest.class);
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java]: Move the LOGGER example to another (its own) package/test. We try to concentrate on one thing only when preparing a showcase.

File: src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+	@Test
+	@Description(value = "Showcase for clicking a random element in the list of urrent job offers.")
+	public void testRandomJobs() {
+		// open the demo page and prepare it for the test
+		openJobPage();
+
+		// check if the heading "Aktuelle Stellenangebote" exists
+//        $(".col-sm-4.tab-content h2").should(matchText("Aktuelle Stellenangebote"));
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java]: remove unneeded comments

File: config/log4j2.properties [hunk 10]
```diff
@@ -0,0 +1,44 @@
+##############################################################################
```
Comment (occupant23) [inline config/log4j2.properties]: remove this file, since there is the same file in the correct place.

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+		
+		// test job offer detail page contains ".job-listing-col.job-col-header" twice
+		// it is unique in this page
+		$$(".job-listing-col.job-col-header").shouldHave(sizeGreaterThanOrEqual(2));
+
+	}
+
+	// a helper function to open the job page //and close the GDPR dialog to avoid
+	// duplicate code
+	private void openJobPage() {
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: move this function into a OpenPageFlow class as static function and call it here:
`OpenPageFlow.openXceptanceJobOffersPage();`

Adapt this also for the other test classes to avoid duplicate code.

**Review by occupant23 — CHANGES_REQUESTED** [review]
Please adapt the requested changes.

File: config/neodymium.properties (hunk 1 above)
Comment (ufjena) [reply config/neodymium.properties]: done

File: config/neodymium.properties (hunk 2 above)
Comment (ufjena) [reply config/neodymium.properties]: done

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java (hunk 3 above)
Comment (ufjena) [reply src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: done

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java (hunk 4 above)
Comment (ufjena) [reply src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: done

File: src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java (hunk 9 above)
Comment (ufjena) [reply src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java]: done

File: config/log4j2.properties (hunk 10 above)
Comment (ufjena) [reply config/log4j2.properties]: done

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java (hunk 5 above)
Comment (ufjena) [reply src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: done

File: src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java (hunk 7 above)
Comment (ufjena) [reply src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java]: done

File: src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java (hunk 8 above)
Comment (ufjena) [reply src/test/java/com/xceptance/neodymium/random/RandomJobOffersTest.java]: done

File: src/test/java/showcase/flows/OpenPageFlow.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+import static com.codeborne.selenide.Selenide.$;
+
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.ex.ElementNotFound;
+
+public class OpenPageFlow
+{
+    public static void openXceptanceJobOffersPage()
+    {
+        // a helper function to open the job page and close the GDPR dialog to avoid duplicate code
```
Comment (occupant23) [inline src/test/java/showcase/flows/OpenPageFlow.java]: move comment above the function (can also changed into JavaDoc)

File: src/test/java/showcase/flows/OpenPageFlow.java [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+        {
```
Comment (occupant23) [inline src/test/java/showcase/flows/OpenPageFlow.java]: remove unneeded brackets (capsuling)

File: src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java [hunk 14]
```diff
@@ -1,56 +0,0 @@
-package showcase.neodymium.tests.pdf;
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java]: Why is this test case removed?

File: src/test/java/showcase/neodymium/tests/logging/LoggingSecondTest.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+import org.apache.logging.log4j.Logger;
+import org.junit.Test;
+
+import io.qameta.allure.Description;
+
+public class LoggingSecondTest
+{
+    private static Logger logger = LogManager.getLogger(LoggingSecondTest.class);
+
+    @Test
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/logging/LoggingSecondTest.java:18]: add a description to explain what is different compared to the first test

File: src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.codeborne.selenide.ElementsCollection;
+import com.xceptance.neodymium.util.NeodymiumRandom;
+
+import io.qameta.allure.Description;
+import showcase.flows.OpenPageFlow;
+
+public class RandomJobOffersTest
+{
+    @Test
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java:30]: add a pre condition that no initial random value is set to ensure that the outcome of the test differs with every run.

File: src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+        final int randomNumber = NeodymiumRandom.nextInt(jobOffersCount) + 1;
+
+        // remember the selected headline
+        final String headline = jobOffersList.get(randomNumber).getText();
+
+        // click randomNumber'th link in the job offers list
+        jobOffersList.get(randomNumber).click();
+
+        // check if job offer detail page contains same headline
+        Assert.assertEquals("The headline is not the same", headline,
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java]: Please use the Selenide to validate the text since this is preferred over Assert `$("").should(exactText(headline));`

File: src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+        // check if the heading "Aktuelle Stellenangebote" exists
+        $(".col-sm-4.tab-content h2").shouldHave(exactText("Aktuelle Stellenangebote"));
+
+        ElementsCollection jobOffersList = $$(".tab-content>.jobs-plain-list>li>a");
+
+        // count the job offers in the list
+        final int jobOffersCount = jobOffersList.size();
+
+        // create a random number between 1 and job offers count in the list
+        final int randomNumber = NeodymiumRandom.nextInt(jobOffersCount) + 1;
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java]: +1 is not needed? as far as I know does random it return a value from 0 to  jobOffersCount-1. 
This matches the interface of `jobOffersList.get()`.

**Review by occupant23 — CHANGES_REQUESTED** [review]

File: src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java (hunk 11 above)
Comment (ufjena) [reply src/test/java/com/xceptance/neodymium/random/FixedRandomTest.java]: done

File: src/test/java/showcase/flows/OpenPageFlow.java (hunk 13 above)
Comment (ufjena) [reply src/test/java/showcase/flows/OpenPageFlow.java]: done

File: src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java (hunk 14 above)
Comment (ufjena) [reply src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java]: by mistake

File: src/test/java/showcase/neodymium/tests/logging/LoggingSecondTest.java (hunk 15 above)
Comment (ufjena) [reply src/test/java/showcase/neodymium/tests/logging/LoggingSecondTest.java:18]: There is no difference. is just a second test class for a second log file.

File: src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java (hunk 16 above)
Comment (ufjena) [reply src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java:30]: done

File: src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java (hunk 18 above)
Comment (ufjena) [reply src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java]: done

File: src/test/java/showcase/flows/OpenPageFlow.java (hunk 12 above)
Comment (ufjena) [reply src/test/java/showcase/flows/OpenPageFlow.java]: done

File: src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java (hunk 17 above)
Comment (ufjena) [reply src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java]: done

File: src/test/resources/log4j2.properties [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+logger.file2.name = ${projectPath}.neodymium.tests.logging.LoggingSecondTest
+logger.file2.level = info
+logger.file2.additivity = true
+logger.file2.appenderRef.file2.ref = FILE2
+### END BLOCK for special logging configuration
+
+### default rootLogger.level = error 
+rootLogger.level = fatal
+rootLogger.appenderRef.file.ref = FILE
+rootLogger.appenderRef.console.ref = STDOUT
```
Comment (occupant23) [inline src/test/resources/log4j2.properties]: add newline at EOF

File: src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java (hunk 16 above)
Comment (occupant23) [reply src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java:30]: Use something like the following otherwise configuring another initValue would not be detected by your pre conditions but still lead to the same result on different runs.
 
`Assert.assertNull("FixedRandomTest: neodymium.context.random.initialValue is set", Neodymium.configuration().initialRandomValue());`

File: src/test/java/showcase/neodymium/tests/logging/LoggingSecondTest.java (hunk 15 above)
Comment (occupant23) [reply src/test/java/showcase/neodymium/tests/logging/LoggingSecondTest.java:18]: Thats a difference and the reason for the second test. Please add this information as a comment.

**Review by occupant23 — CHANGES_REQUESTED** [review]

File: src/test/java/showcase/neodymium/tests/logging/LoggingSecondTest.java (hunk 15 above)
Comment (ufjena) [reply src/test/java/showcase/neodymium/tests/logging/LoggingSecondTest.java:18]: done

File: src/test/resources/log4j2.properties (hunk 19 above)
Comment (ufjena) [reply src/test/resources/log4j2.properties]: done

File: src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java (hunk 16 above)
Comment (ufjena) [reply src/test/java/showcase/neodymium/tests/random/RandomJobOffersTest.java:30]: done

**Comment by occupant23** [comment]
@ufjena :

Please let the new test cases extend `AbstractTest`:
- LoggingFirstTest
- LoggingSecondTest
- FixedRandomTest
- RandomJobOffersTest

Add the `@SuppressBrowsers` annotation to the following case to avoid opening a browser:
- LoggingFirstTest
- LoggingSecondTest
- FixedRandomTest

File: src/test/java/showcase/flows/OpenPageFlow.java [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+        catch (ElementNotFound e)
+        {
+            overlayIsVisible = false;
+        }
+
+        if (overlayIsVisible)
+        {
+            $(".btn-link").click();
+            $(".btn-link").shouldBe(hidden);
+            Selenide.refresh();
```
Comment (occupant23) [inline src/test/java/showcase/flows/OpenPageFlow.java]: refresh should not be needed at this place

**Review by occupant23 — APPROVED** [review]
